### PR TITLE
tests: extract integration tests helpers so separate packages for reuse

### PR DESCRIPTION
**Pull request by pmalek**
**What this PR does / why we need it**:

This PR extracts many of the integration test helpers to separate packages so that
- they can be reused
- they don't rely on shared variables e.g. in `utils_test.go`

The latter point still holds true for some of the variables but let's iterate on this in another this PR (this is already rather large).

### CODE REVIEW

File: test/consts/exit_codes.go [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+	ExitCodeCleanupFailed = 103
+
+	// ExitCodeEnvSetupFailed is a generic exit code that can be used as a fallback for general
+	// problems setting up the testing environment and/or cluster.
+	ExitCodeEnvSetupFailed = 104
+
+	// ExitCodeCentCreateLogger is a POSIX compliant exit code for the test suite to indicate
+	// that a failure occurred when trying to create a logger for the test suite.
+	ExitCodeCantCreateLogger = 105
+)
```
Comment (czeslavo) [inline test/consts/exit_codes.go:35]: Do you have any idea how these specific exit codes are meant to help us? As I can see, the `exitOnErrWithCode` function that they are used in pair with, logs the error to stderr, which should be visible in the CI. I don't understand how using specific error codes will give us more context than the error log. 🤔

Comment (czeslavo) [reply test/consts/exit_codes.go:35]: I understand it was just moved so don't bother to change it in this PR, but I'm just curious if I'm missing something.

File: test/integration/ingress_test.go [hunk 2]
```diff
@@ -38,15 +39,15 @@ var ingressClassMutex = sync.Mutex{}
 func TestIngressEssentials(t *testing.T) {
 	t.Parallel()
 
-	t.Log("locking IngressClass management")
+	t.Log("locking consts.IngressClass management")
```
Comment (czeslavo) [inline test/integration/ingress_test.go:42]: Renaming in this file went wrong 😛 (`consts.IngressClass` in log messages).

File: test/internal/helpers/ingress.go [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+
+	"github.com/kong/kubernetes-testing-framework/pkg/clusters"
+	netv1 "k8s.io/api/networking/v1"
+	netv1beta1 "k8s.io/api/networking/v1beta1"
+	"k8s.io/apimachinery/pkg/runtime"
+)
+
+// -----------------------------------------------------------------------------
+// Ingress Helpers
+// -----------------------------------------------------------------------------
```
Comment (czeslavo) [inline test/internal/helpers/ingress.go:14]: I've been wondering how these "section" comments are useful. They typically are used when the content below them should be extracted as a separate file IMHO. In this case it's already a separate file and the comment duplicates the package/file pattern. Do you mind removing it?

File: test/internal/helpers/kong.go [hunk 4]
```diff
@@ -0,0 +1,52 @@
...
+	"net/url"
+	"os"
+
+	"github.com/blang/semver/v4"
+	"github.com/kong/go-kong/kong"
+)
+
+// -----------------------------------------------------------------------------
+// Testing Utility Functions - Kong
+// -----------------------------------------------------------------------------
```
Comment (czeslavo) [inline test/internal/helpers/kong.go:16]: Ditto

File: test/internal/testenv/testenv.go [hunk 5]
```diff
@@ -0,0 +1,94 @@
...
+}
+
+// KongPullPassword is the Docker password to use for the Kong image pull secret.
+func KongPullPassword() string {
+	return os.Getenv("TEST_KONG_PULL_PASSWORD")
+}
+
+// KongEnterpriseEnabled enables Enterprise-specific tests when set to "true".
+func KongEnterpriseEnabled() string {
+	return os.Getenv("TEST_KONG_ENTERPRISE")
```
Comment (czeslavo) [inline test/internal/testenv/testenv.go:59]:
```suggestion
func KongEnterpriseEnabled() bool {
	return os.Getenv("TEST_KONG_ENTERPRISE") == "true"
```

File: test/internal/helpers/namespace.go [hunk 6]
```diff
@@ -0,0 +1,29 @@
...
+	"github.com/kong/kubernetes-testing-framework/pkg/clusters"
+	"github.com/kong/kubernetes-testing-framework/pkg/environments"
+	"github.com/stretchr/testify/assert"
+	"github.com/stretchr/testify/require"
+	corev1 "k8s.io/api/core/v1"
+)
+
+// -----------------------------------------------------------------------------
+// Testing Utility Functions - Namespaces
+// -----------------------------------------------------------------------------
```
Comment (czeslavo) [inline test/internal/helpers/namespace.go:16]: Ditto

File: test/internal/helpers/ingress.go (hunk 3 above)
Comment (pmalek) [reply test/internal/helpers/ingress.go:14]: 👍

File: test/integration/ingress_test.go (hunk 2 above)
Comment (pmalek) [reply test/integration/ingress_test.go:42]: 👍

**Review by czeslavo — APPROVED** [review]

File: test/consts/exit_codes.go (hunk 1 above)
Comment (pmalek) [reply test/consts/exit_codes.go:35]: Perhaps to be POSIX compliant just like mentioned in the comments :)

I believe these are a nice touch for someone running the tests and inspecting the return code programmatically 🤷

**Pull request merged**